include/net/net_context.h

```diff
@@ -222,6 +228,15 @@ struct net_context {
 #endif
 	} options;
 
+#if defined(CONFIG_NET_TLS) || defined(CONFIG_NET_DTLS)
+#if defined(CONFIG_MBEDTLS)
+	struct {
+		mbedtls_ssl_context ssl;
```

Comment:
This makes any and every network context waste space on mbedTLS structures, regardless of whether TLS is actually used with it or not. And mbedTLS structures are quite big in general.

Reply:
In practice this needs to be a pointer that is "allocated" to this context only when needed.

Reply:
Yes. This patch just intends to show(el) TLS support placed somewhere, it's not the last word on the issue.

Reply:
We should have a separate memory area for TLS data and access them using net_context as key. But as said, this is so far just a prototype. Next step is to have TLS set up properly.

Reply:
> In practice ...

In another practice, nothing of that is needed. Because TLS is layered on top of UDP/TCP context, so TLS needs to point to such a context, not the other way around.

But memory management of TLS-related structures is indeed one of the hardest technical problems of this "setockopt, push-under-the-socket-layer" approach. Zstream deals with it in a very smart way - in no way at all. It never allocates such structures, but instead always receives them via dependency injection pattern. Allocation is left to users of API. Yes, that means they have to do it. But that also means they have ultimate flexibility with it. For example, for clients, which require just one TLS structure, it can be allocated statitically, with zero additional management overhead. For servers, it's more complicated, but still offers ultimate flexibility. For example, allocation can use kernel k_mem_slab for a fixed number of connections for guaranteed service, and if that overflows, switch to allocation from heap (for best effort service of larger number of conns). All that without adding any overhead to Zephyr kernel.